### PR TITLE
[4.0] nova: Add option to modify scheduler_default_filters

### PR DESCRIPTION
**Pull request by jsuchome**
In some cases user needs to modify default filters, e.g. remove
DiskFilter when instances are only booted from volumes (bsc#1080997)

See https://github.com/crowbar/crowbar-openstack/pull/1578 for master version

### CODE REVIEW

**Comment by jsuchome** [comment]
[P2 - Bug 1080997 - nova.compute.resource_tracker issue with Netapp backend](https://trello.com/c/dVAr0rpH/696-p2-bug-1080997-novacomputeresourcetracker-issue-with-netapp-backend)

**Comment by jsuchome** [comment]
@JanZerebecki I think it does not need to go *first* into master - we still have quite some time to figure out upgrade process, so there's no harm when it waits few days...

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -34,9 +34,20 @@ reserved_host_memory_mb = <%= @reserved_host_memory %>
...
-<% if @has_itxt %>scheduler_default_filters = RetryFilter,AvailabilityZoneFilter,RamFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,TrustedFilter<% end %>
+<% if @use_baremetal_filters %>
+scheduler_use_baremetal_filters = true
+<% end -%>
+<% if @has_itxt %>
+scheduler_available_filters = nova.scheduler.filters.standard_filters
+<% if @default_filters.empty? %>
+scheduler_default_filters = RetryFilter,AvailabilityZoneFilter,RamFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,TrustedFilter
+<% else %>
+scheduler_default_filters = @default_filters
```
Comment (jsuchome) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: oh, this is wrong

**Comment by JanZerebecki** [comment]
Looks good, good catch.

**Comment by jsuchome** [comment]
@rsalevsky @nicolasbock another review round please?

**Pull request merged**